### ISSUE TITLE
Homing weapons refuse to autofire without a target lock

### ISSUE THREAD

**Issue by Bladewood** [endless-sky/endless-sky#2715]
Post-individual turret aiming thing. (It may have been pre- as well, I don't know.)
This happens with Korath grab-strike turrets and Pug seekers. Seekers won't fire even if an enemy is right in front of them, and grab-strikes will aim at enemies but will not fire.

(I made this an issue not because I don't personally like it, but because it is inconsistent with other workings.)

**Comment by tehhowch** [comment]
AI.cpp::Autofire()
```
	for(const Hardpoint &weapon : ship.Weapons())
	{
		++index;
		// Skip weapons that are not ready to fire. Also skip homing weapons if
		// no target is selected, and secondary weapons if only firing primaries.
		if(!weapon.IsReady() || (!currentTarget && weapon.IsHoming()))
			continue;
		if(!secondary && weapon.GetOutfit()->Icon())
			continue;
		if(beFrugal && weapon.GetOutfit()->Ammo())
			continue;
```
[This section has not been touched since individual turret targeting](https://github.com/endless-sky/endless-sky/blame/master/source/AI.cpp#L2127)

The issue here is that the launched projectiles from homing weapons acquire their [target from the ship that fired](https://github.com/endless-sky/endless-sky/blame/master/source/Projectile.cpp#L41) it, and not the turret that fired it.
We could extend the logic in Projectile.cpp to do some aiming if the projectile is homing and there is no targeted ship.

**Comment by endless-sky** [comment]
Fixed in 957f6c0, by allowing any homing weapon that doesn't consume ammo to revert to "dumb" auto-firing if no target is selected.

Note that seekers and grab-strikes are both relatively slow-moving projectiles, so with no target selected they tend to miss a lot. But, that commit will at least make them fire.
